src/ray/core_worker/transport/direct_actor_transport.cc

```diff
@@ -92,6 +92,19 @@ Status CoreWorkerDirectActorTaskSubmitter::SubscribeActorUpdates() {
     } else {
       // Remove rpc client if it's dead or being reconstructed.
       rpc_clients_.erase(actor_id);
+      // If this actor is permanantly dead and there're pending requests, treat
+      // the pending tasks as failed.
+      if (actor_data.state() == ActorTableData::DEAD &&
+          pending_requests_.count(actor_id) > 0) {
+        auto &requests = pending_requests_[actor_id];
```

Comment:
Would prefer range-based for loop and then a call to `pending_requests_.clear()`, more concise. Also don't need the `pending_requests.count(actor_id)` check.

Reply:
without `pending_requests.count(actor_id)`, `pending_requests_[actor_id]` will construct an empty list.